boards/airbourne/airbourne_board_config_manager.cpp

```diff
@@ -0,0 +1,261 @@
+  ConfigManager &cm = RF_->config_manager_;
+  resp.reboot_required = false;
+  resp.successful = false;
+  char *error_message = reinterpret_cast<char*>(resp.message); // Because it wasn't converting implicitly
+
+  if(device >= Configuration::DEVICE_COUNT)
+  {
+    strcpy(error_message, "Device not found");
+    return resp;
+  }
+
+  if(config > max_configs[device])
+  {
+    strcpy(error_message, "Configuration not found");
+    return resp;
+  }
+  if(config == cm[device])
+  {
+    strcpy(error_message, "Configuration already set. No change required");
+    resp.successful = true;
+    resp.reboot_required = false;
+    return resp;
+  }
+
+  revo_port port = get_port(device, config);
+  device_t conflict_device = Configuration::DEVICE_COUNT;
+
+  // While a config may conflict with multiple devices, this will only report one
+  switch(device)
+  {
+  case Configuration::RC:
+    if(config ==0) // PPM is not known to conflict with anything
+      break;
+  case Configuration::SERIAL:
+  case Configuration::GNSS:
+    if(port != NO_PORT)
+      for(device_t other_device{static_cast<device_t>(0)}; other_device != Configuration::DEVICE_COUNT; ++other_device)
+        if(port == get_port(other_device, cm[other_device]) && (other_device !=Configuration::RC || cm[Configuration::RC]!=0)) // RC over PPM does not conflict with UART, even though both use the same port
+        {
+          conflict_device = other_device;
+          break;
+        }
+    break;
+  case Configuration::AIRSPEED:
+  case Configuration::SONAR:
+    if(cm[Configuration::GNSS] == 3)
+      conflict_device = Configuration::GNSS;
+    if(cm[Configuration::SERIAL] == 3)
+      conflict_device = Configuration::SERIAL;
+    break;
+  default:
+    break;
+  }
+  if(conflict_device != Configuration::DEVICE_COUNT)
+  {
+    switch(conflict_device)
+    {
+    case Configuration::SERIAL:
+      strcpy(error_message, "Port is used by serial.");
+      break;
+    case Configuration::RC:
+      strcpy(error_message, "Port is used by RC.");
+      break;
+    case Configuration::AIRSPEED:
+      strcpy(error_message, "Port is used by airspeed sensor.");
+      break;
+    case Configuration::GNSS:
+      strcpy(error_message, "Port is used by GNSS receiver.");
+      break;
+    case Configuration::SONAR:
+      strcpy(error_message, "Port is used by sonar sensor.");
+      break;
+    // At the time of this writing, the below are incapable of conflicts
+    case Configuration::BATTERY_MONITOR:
+      strcpy(error_message, "Port is used by battery monitor.");
+      break;
+    case Configuration::BAROMETER:
+      strcpy(error_message, "Port is used by barometer.");
+      break;
+    case Configuration::MAGNETOMETER:
+      strcpy(error_message, "Port is used by magnetometer.");
+      break;
+    default:
+      strcpy(error_message, "Other error.");
+      break;
+    }
+    return resp;
+  }
+  resp.successful = true;
+  resp.reboot_required = true;
+  resp.message[0]=0;
```

Comment:
The response is filled with failure data (resp.successful = false, resp.reboot_required=false), because there are many more cases where it fails then where it succeeds. This is settings the values for successful changes.
If you were asking specifically about `resp.message[0]=0;`, that is putting a null byte at the first entry in the string, so that it is treated as a zero length string.